File: backend_python/app/services/booking_service.py
"""
Booking service
"""
from datetime import datetime
from app.models.booking import Booking, BookingStatus
from app.middleware.error_handler import HttpError


class BookingService:
    """Booking operations"""
    
    @staticmethod
    async def create_booking(
        user_id: str,
        data: dict,
    ) -> Booking:
        """Create a new booking"""
        booking = Booking(
            userId=user_id,
            serviceId=data["serviceId"],
            scheduledDate=datetime.fromisoformat(data["scheduledDate"].replace("Z", "+00:00")),
            status=BookingStatus.CONFIRMED,
        )
        await booking.insert()
        return booking

File: backend/src/modules/booking/booking.service.ts
import type { PrismaClient } from '@prisma/client';

export function createBooking(
  prisma: PrismaClient,
  userId: string,
  data: { serviceId: string; scheduledDate: string },
) {
  return prisma.booking.create({
    data: {
      userId,
      serviceId: data.serviceId,
      scheduledDate: new Date(data.scheduledDate),
    },
  });
}

